packages/web-components/fast-foundation/src/combobox/combobox.ts

```diff
@@ -608,26 +586,46 @@ export class FASTCombobox extends FormAssociatedCombobox {
     /**
      * Calculate and apply listbox positioning based on available viewport space.
      *
-     * @param force - direction to force the listbox to display
      * @public
      */
     public setPositioning(): void {
-        const currentBox = this.getBoundingClientRect();
-        const viewportHeight = window.innerHeight;
-        const availableBottom = viewportHeight - currentBox.bottom;
-
-        this.position = this.forcedPosition
-            ? this.positionAttribute
-            : currentBox.top > availableBottom
-            ? SelectPosition.above
-            : SelectPosition.below;
-
-        this.positionAttribute = this.forcedPosition
-            ? this.positionAttribute
-            : this.position;
-
-        this.maxHeight =
-            this.position === SelectPosition.above ? ~~currentBox.top : ~~availableBottom;
+        if (this.$fastController.isConnected) {
+            this.cleanup = autoUpdate(this, this.listbox, async () => {
+                const { middlewareData, x, y } = await computePosition(
+                    this,
+                    this.listbox,
+                    {
+                        placement: "bottom",
+                        strategy: "fixed",
+                        middleware: [
+                            flip(),
+                            size({
+                                apply: ({ availableHeight, rects }) => {
+                                    Object.assign(this.listbox.style, {
+                                        maxHeight: `${availableHeight}px`,
+                                        width: `${rects.reference.width}px`,
+                                    });
+                                },
+                            }),
+                            hide(),
+                        ],
+                    }
+                );
+
+                if (middlewareData.hide?.referenceHidden) {
+                    this.open = false;
+                    this.cleanup();
+                    return;
+                }
+
+                Object.assign(this.listbox.style, {
+                    position: "fixed",
+                    top: "0",
+                    left: "0",
+                    transform: `translate(${x}px, ${y}px)`,
+                });
+            });
```

Comment:
I'd love to see us look into abstracting this into a utility which could be shared rather than repeated once we have a good baseline for implementation.